src/main/java/net/minecraftforge/client/model/generators/BlockModelBuilder.java

```diff
@@ -16,9 +25,374 @@
+        {
+            Preconditions.checkNotNull(transformation, "Transformation must not be null");
+            this.translation = transformation.getTranslation();
+            this.leftRotation = transformation.getLeftRotation();
+            this.rightRotation = transformation.getRightRotation();
+            this.scale = transformation.getScale();
+            return this;
+        }
+
+        /**
+         * Sets the origin of the root transform.
+         *
+         * @param origin the origin vector
+         * @return this builder
+         * @throws NullPointerException if {@code origin} is {@code null}
+         */
+        public RootTransformBuilder origin(Vector3f origin)
+        {
+            this.originVec = Preconditions.checkNotNull(origin, "Origin must not be null");
+            this.origin = null;
+            return this;
+        }
+
+        /**
+         * Sets the origin of the root transform.
+         *
+         * @param origin the origin name
+         * @return this builder
+         * @throws NullPointerException if {@code origin} is {@code null}
+         * @throws IllegalArgumentException if {@code origin} is not {@code center}, {@code corner} or {@code opposing-corner}
+         */
+        public RootTransformBuilder origin(TransformOrigin origin)
+        {
+            this.origin = Preconditions.checkNotNull(origin, "Origin must not be null");
+            this.originVec = null;
+            return this;
+        }
+
+        /**
+         * Finish configuring the parent builder
+         * @return the parent block model builder
+         */
+        public BlockModelBuilder end()
+        {
+            return BlockModelBuilder.this;
+        }
+
+        public JsonObject toJson() {
+            // Write the transform to an object
+            JsonObject transform = new JsonObject();
+
+            if (!translation.equals(Vector3f.ZERO))
+            {
+                transform.add("translation", writeVec3(translation));
+            }
+
+            if (!scale.equals(ONE))
+            {
+                transform.add("scale", writeVec3(scale));
+            }
+
+            if (!leftRotation.equals(Quaternion.ONE))
+            {
+                transform.add("rotation", writeQuaternion(leftRotation));
+            }
+
+            if (!rightRotation.equals(Quaternion.ONE))
+            {
+                transform.add("post_rotation", writeQuaternion(rightRotation));
+            }
+
+            if (origin != null)
+            {
+                transform.addProperty("origin", origin.getSerializedName());
+            }
+            else if (originVec != null && !originVec.equals(Vector3f.ZERO))
+            {
+                transform.add("origin", writeVec3(originVec));
+            }
+
+            return transform;
+        }
+
+        public enum TransformOrigin implements StringRepresentable
+        {
+            CENTER(new Vector3f(.5f, .5f, .5f), "origin"),
```

Comment:
Why the name of origins was changed? Before the CENTER was "center" and OPPOSING_CORNER was "opposing-corner". Now every model that used old names inside .json files is broken in 43.1.9+

Reply:
These are the constants in code, however the third argument on the line highlighted sets the name that is used during serialisation, therefore this will not break compat. However if you are having issues with this PR, opening an issue with a bit more info might be helpful

Reply:
Ooh I have spotted the underscore issue, I'll mention it in forgecord to see if someone's free to issue a fix

Reply:
If anyone had "origin": "center" before it will parse it, but fail because now "center" is called "origin". And yes, same for underscore issue.

Reply:
Yeah just clocked that too, I'm reading on phone right now and I missed it, was going off of memory. I'll try and get this sorted out ASAP. Thanks for reporting this to me, very sorry

Reply:
All good :)